app/src/main/java/net/squanchy/service/repository/firebase/FirestoreSpeakerRepository.kt

```diff
@@ -0,0 +1,25 @@
+package net.squanchy.service.repository.firebase
+
+import io.reactivex.Observable
+import net.squanchy.service.firestore.FirestoreDbService
+import net.squanchy.service.firestore.toSpeaker
+import net.squanchy.service.repository.SpeakerRepository
+import net.squanchy.speaker.domain.view.Speaker
+import net.squanchy.support.lang.Checksum
+
+class FirestoreSpeakerRepository(
+    private val dbService: FirestoreDbService,
+    private val checksum: Checksum
+) : SpeakerRepository {
+
+    override fun speakers(): Observable<List<Speaker>> {
+        return dbService.speakers()
+            .map { it.map { it.toSpeaker(checksum) } }
+    }
+
+    override fun speaker(speakerId: String): Observable<Speaker> {
+        return dbService.speakers()
```

Comment:
this has to go with a `dbService.speaker(speakerId)`, we have firestore so we can do querying :)